src/client.rs

```diff
@@ -1675,6 +1678,20 @@ where
             Some(message) => message,
             None => &self.buffer,
         };
+        // TODO query sample rate from config
+        let should_parse_query = pool.settings.query_parser_enabled
+            && !server.in_transaction()
```

Comment:
Can we also make sure it's not a begin statement we're caching or a set statement? Ideally we just check for `select from`